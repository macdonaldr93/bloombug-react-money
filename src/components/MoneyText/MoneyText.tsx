import React, { CSSProperties, FC, useMemo } from 'react';
import { CurrencyCode } from '@bloombug/money';

import { useMoney } from '../../hooks';

export interface MoneyTextProps
  extends Omit<Partial<Intl.NumberFormatOptions>, 'style' | 'currency'> {
  className?: string;
  currency?: CurrencyCode | null;
  fractional?: bigint | number | string;
  locale?: string | string[];
  style?: CSSProperties;
}

export const MoneyText: FC<MoneyTextProps> = ({
  locale,
  fractional,
  currency,
  className,
  style,
  ...formatOptions
}) => {
  const { Money } = useMoney();

  const formattedMoney = useMemo(
    () => Money(fractional, currency).format(locale, formatOptions),
    [Money, fractional, currency, locale, formatOptions]
  );

  return (
    <span className={className} style={style}>
      {formattedMoney}
    </span>
  );
};
