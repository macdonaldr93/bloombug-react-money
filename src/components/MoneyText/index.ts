export { MoneyText, MoneyTextProps } from './MoneyText';
