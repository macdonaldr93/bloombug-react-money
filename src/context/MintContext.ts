import { createContext } from 'react';
import { Mint } from '@bloombug/money';

export interface IMintContext {
  mint: Mint;
}

export const MintContext = createContext<IMintContext>({ mint: new Mint() });
