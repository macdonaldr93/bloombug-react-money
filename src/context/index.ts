export { IMintContext, MintContext } from './MintContext';
export { MintProvider, MintProviderProps } from './MintProvider';
