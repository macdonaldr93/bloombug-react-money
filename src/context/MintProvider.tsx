import React, { FC, ReactChild } from 'react';
import { Mint } from '@bloombug/money';

import { MintContext } from './MintContext';

export interface MintProviderProps {
  children: ReactChild;
  mint: Mint;
}

export const MintProvider: FC<MintProviderProps> = ({ children, mint }) => {
  return (
    <MintContext.Provider value={{ mint }}>{children}</MintContext.Provider>
  );
};
