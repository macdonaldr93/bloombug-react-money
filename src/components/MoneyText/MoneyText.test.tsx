import React from 'react';
import * as ReactDOM from 'react-dom';
import { CAD, Mint } from '@bloombug/money';
import currencies from '@bloombug/money/iso-currencies.json';

import { MintProvider } from '../../context';
import { MoneyText } from './MoneyText';

describe('<MoneyText />', () => {
  let div: HTMLDivElement;

  beforeEach(() => {
    div = document.createElement('div');
  });

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(div);
  });

  it('returns default format money', () => {
    ReactDOM.render(<MoneyText fractional={100} />, div);

    expect(div).toMatchSnapshot();
  });

  it('returns formatted money', () => {
    const mint = new Mint({ currencies });

    ReactDOM.render(
      <MintProvider mint={mint}>
        <MoneyText fractional={100} currency={CAD} locale="fr-FR" />
      </MintProvider>,
      div
    );

    expect(div).toMatchSnapshot();
  });
});
