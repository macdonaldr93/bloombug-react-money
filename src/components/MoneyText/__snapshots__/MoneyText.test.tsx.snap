// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoneyText /> returns default format money 1`] = `
<div>
  <span>
    US$1.00
  </span>
</div>
`;

exports[`<MoneyText /> returns formatted money 1`] = `
<div>
  <span>
    1,00Â $CA
  </span>
</div>
`;
